### PR TITLE
OSLQuery improvements

### PR DESCRIPTION
**Pull request by lgritz**
* OSLQuery::Parameter -- add copy/move constructors

* Make the ShadingSystem side of OSLQuery correctly report default vals.

### CODE REVIEW

File: src/include/OSL/oslquery.h [hunk 1]
```diff
@@ -75,10 +75,10 @@ class OSLQUERYPUBLIC OSLQuery {
         std::vector<ustring> fields;     ///< Names of this struct's fields
         ustring structname;              ///< Name of the struct
         std::vector<Parameter> metadata; ///< Meta-data about the param
-        Parameter ()
-            : isoutput(false), validdefault(false), varlenarray(false),
-              isstruct(false), isclosure(false), data(NULL)
-        { }
+
+        Parameter () {}
```
Comment (fpsunflower) [inline src/include/OSL/oslquery.h]: Is there any need to declare these? Don't you just want the "default" behavior?

Comment (lgritz) [reply src/include/OSL/oslquery.h]: For the copy and move constructors, it's important that the `data` pointer refers to ITS OWN default field, not a straight pointer copy from the source. So default ctrs won't work.

Comment (lgritz) [reply src/include/OSL/oslquery.h]: ok with this?

Comment (fpsunflower) [reply src/include/OSL/oslquery.h]: Ah I missed that detail.

LGTM!

**Pull request merged**